applications/dashboard/controllers/api/AbstractApiController.php

```diff
@@ -76,11 +79,28 @@ public function getUserFragmentSchema() {
                 'name:s' => 'The username of the user.',
                 'photoUrl:s' => 'The URL of the user\'s avatar picture.',
                 'dateLastActive:dt|n' => 'Time the user was last active.',
+                "permissions:a?" => $this->getPermissionFragment(),
```

Comment:
Was there a particular use case you were imagining where someone would have need to the current user and _not_ have need of their permissions?

If not than this should be an always present field instead of an expand field.

Reply:
> Was there a particular use case you were imagining where someone would have need to the current user and not have need of their permissions?

Definitely. Maybe if it were going to *only* be global permissions, including them by default would probably be fine. However, some sites have notoriously complicated permissions setups, which can quickly ballon the size of the response. If I have 1,000 categories and custom permissions for each, my "permissions" entry is going to be huge. If all I need to do is build a simple "me" box or get the current user's ID or name, a call to this endpoint will suffice. I don't want to unnecessarily bloat that response with an excessive amount of unnecessary data. Having the consumer explicitly request permissions makes more sense to me.

Reply:
So is this giving me "all" permissions? Because that's going to be an immediate issue. My understanding is that were going to put __only__ global permissions here, and that resource specific permissions would come down with the resource, or in a separate API call on the resource.

So if we end up needing category specific permissions we would expand permissions in a request to the permissions.

---

Maybe I put this comment on the wrong line. My thought process is:

- `/users/me` will __always__ have permissions
- The permissions on `/users/me` will __only__ be global permissions.